galleries/examples/subplots_axes_and_figures/secondary_axis.py

```diff
@@ -96,48 +95,50 @@ def one_over(x):
 plt.show()
 
 # %%
-# Sometime we want to relate the axes in a transform that is ad-hoc from
-# the data, and is derived empirically.  In that case we can set the
-# forward and inverse transforms functions to be linear interpolations from the
-# one data set to the other.
+# Sometime we want to relate the axes in a transform that is ad-hoc from the data, and
+# is derived empirically. Or, one axis could be a complicated nonlinear function of the
+# other. In these cases we can set the forward and inverse transform functions to be
+# linear interpolations from the one set of independent variables to the other.
 #
 # .. note::
 #
 #   In order to properly handle the data margins, the mapping functions
 #   (``forward`` and ``inverse`` in this example) need to be defined beyond the
-#   nominal plot limits.
-#
-#   In the specific case of the numpy linear interpolation, `numpy.interp`,
-#   this condition can be arbitrarily enforced by providing optional keyword
-#   arguments *left*, *right* such that values outside the data range are
-#   mapped well outside the plot limits.
+#   nominal plot limits. This condition can be enforced by extending the
+#   interpolation beyond the plotted values, both to the left and the right,
+#   see ``x1n`` and ``x2n`` below.
 
 fig, ax = plt.subplots(layout='constrained')
-xdata = np.arange(1, 11, 0.4)
-ydata = np.random.randn(len(xdata))
-ax.plot(xdata, ydata, label='Plotted data')
-
-xold = np.arange(0, 11, 0.2)
-# fake data set relating x coordinate to another data-derived coordinate.
-# xnew must be monotonic, so we sort...
-xnew = np.sort(10 * np.exp(-xold / 4) + np.random.randn(len(xold)) / 3)
-
-ax.plot(xold[3:], xnew[3:], label='Transform data')
-ax.set_xlabel('X [m]')
+x1_vals = np.arange(2, 11, 0.4)
+# second independent variable is a nonlinear function of the other.
+# this simple example can be more easily handled without interpolation, and is done
+# this way only for pedagogical purposes.
+x2_vals = x1_vals ** 2
+ydata = 50.0 + 20 * np.random.randn(len(x1_vals))
+ax.plot(x1_vals, ydata, label='Plotted data')
+ax.plot(x1_vals, x2_vals, label=r'$x_2(x_1)$')
+ax.set_xlabel(r'$x_1$')
 ax.legend()
 
+# need to define mapped values outside of plotted range to ensure the secondary
+# axis is plotted correctly
+x1n = np.concatenate(([0.0], x1_vals, [20]))
+x2n = np.concatenate(([0.0], x2_vals, [20**2]))
```

Comment:
```suggestion
x1n = np.concatenate(([-1e10, 0], x1_vals, [20, 1e10]))
x2n = np.concatenate(([-1e10, 0], x2_vals, [20**2, 1e10**2]))
```
Might as well make these very large so there is never a problem?